presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveTableStatistics.java

```diff
@@ -756,6 +773,39 @@ public void testAnalyzeForTablePartitionedByBigint()
                 row("p_regionkey", null, 1.0, 0.0, null, "2", "2"),
                 row("p_comment", 351.0, 5.0, 0.0, null, null, null),
                 row(null, null, null, null, 5.0, null, null));
+
+        String filterCommentAllPartitions = "EXPLAIN (TYPE IO) SELECT p_regionKey FROM " + tableNameInDatabase + " WHERE p_comment='foo'";
+        assertEstBytesScanned(filterCommentAllPartitions, 1197);
+    }
+
+    @Test
+    @Requires({NationPartitionedByBigintTable.class, UnpartitionedNationTable.class})
+    public void testEstBytesScannedAcrossJoinedTables()
+    {
+        String tableName1InDatabase = mutableTablesState().get(NATION.getName()).getNameInDatabase();
+        String tableName2InDatabase = mutableTablesState().get(NATION_PARTITIONED_BY_BIGINT_REGIONKEY.getName()).getNameInDatabase();
+
+        String showStatsWholeTable1 = "SHOW STATS FOR " + tableName1InDatabase;
+        String showStatsWholeTable2 = "SHOW STATS FOR " + tableName2InDatabase;
+        assertThat(query("ANALYZE " + tableName1InDatabase)).containsExactly(row(25));
+        assertThat(query("ANALYZE " + tableName2InDatabase)).containsExactly(row(15));
+
+        assertThat(query(showStatsWholeTable1)).containsOnly(
+                row("n_nationkey", null, 25.0, 0.0, null, "0", "24"),
+                row("n_name", 177.0, 25.0, 0.0, null, null, null),
+                row("n_regionkey", null, 5.0, 0.0, null, "0", "4"),
+                row("n_comment", 1857.0, 25.0, 0.0, null, null, null),
+                row(null, null, null, null, 25.0, null, null));
+
+        assertThat(query(showStatsWholeTable2)).containsOnly(
+                row("p_nationkey", null, 5.0, 0.0, null, "1", "24"),
+                row("p_name", 109.0, 5.0, 0.0, null, null, null),
+                row("p_regionkey", null, 3.0, 0.0, null, "1", "3"),
+                row("p_comment", 1197.0, 5.0, 0.0, null, null, null),
+                row(null, null, null, null, 15.0, null, null));
+
+        String joinQuery = "EXPLAIN (TYPE IO) SELECT n_name, p_name FROM " + tableName1InDatabase + " JOIN " + tableName2InDatabase + " ON (n_nationkey = p_nationkey) WHERE n_comment='foo'";
```

Comment:
use `String.format`

Reply:
note: when using `String.format` we static import the method